deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/UrlTree.kt

```diff
@@ -52,27 +55,77 @@ open class TreeNode(open val id: String, val type: UByte, open val placeholder:
         return resultByteArray
     }
 
+    /**
+     * Use a UByte for [flag] so that we can easily specify and read different
+     * combinations/permutations of flags with a bit mask.
+     */
+    enum class NodeMetadata(val flag: UByte) {
```

Comment:
I think @rossbacher originally intended ubiquitous UByte usage to get an extra bit of space per byte.

Reply:
imo since we don't need the extra bit yet, it would be better to not use an experimental api. We can also switch to that down the road.

Reply:
We also use `UByte`s throughout the new (in 2020, since I added Kotlin support everywhere in DLD) Kotlinized parts of DLD, so I think the conversation here expands to the rest of the library. Slacked you and Rossi about this.